File: data/derived_cot/rq2_eval/translation/qwen/_converted_code/java_to_py/FitnessTracker.py
class FitnessTracker:

    def __init__(self, height, weight, age, sex):
        self.height = float(height)
        self.weight = float(weight)
        self.age = int(age)
        self.sex = str(sex).lower()
        self.BMI_std = [[20.0, 25.0], [19.0, 24.0]]

    def get_BMI(self):
        return self.weight / self.height ** 2

    def condition_judge(self):
        try:
            bmi = self.get_BMI()
            index = 0 if self.sex == 'male' else 1
            upper, lower = self.BMI_std[index]
            if bmi > upper:
                return 1
            elif bmi < lower:
                return -1
            else:
                return 0
        except TypeError:
            raise ValueError('Invalid input types. Height, weight, age must be numbers, and sex must be a string.')

    def calculate_calorie_intake(self):
        try:
            bmr = 10 * self.weight + 6.25 * self.height - 5 * self.age + (5 if self.sex == 'male' else -161)
            condition = self.condition_judge()
            if condition == 1:
                return bmr * 1.2
            elif condition == -1:
                return bmr * 1.6
            else:
                return bmr * 1.4
        except TypeError:
            raise ValueError('Invalid input types. Height, weight, age must be numbers, and sex must be a string.')

File: data/derived_cot/rq2_eval/translation/r1/_converted_code/py_to_cpp/FitnessTracker.cpp
#include <map>
#include <vector>
#include <string>

class FitnessTracker {
private:
    double height;
    double weight;
    int age;
    std::string sex;
    std::map<std::string, std::vector<double>> BMI_std_map;

public:
    FitnessTracker(double h, double w, int a, std::string s) 
        : height(h), weight(w), age(a), sex(s) {
        BMI_std_map["male"] = {20.0, 25.0};
        BMI_std_map["female"] = {19.0, 24.0};
    }

    double get_BMI() {
        return weight / (height * height);
    }

    int condition_judge() {
        double BMI = get_BMI();
        std::vector<double> BMI_range;
        if (sex == "male") {
            BMI_range = BMI_std_map["male"];
        } else {
            BMI_range = BMI_std_map["female"];
        }
        if (BMI > BMI_range[1]) {
            return 1;
        } else if (BMI < BMI_range[0]) {
            return -1;
        } else {
            return 0;
        }
    }

    double calculate_calorie_intake() {
        double BMR;
        if (sex == "male") {
            BMR = 10 * weight + 6.25 * height - 5 * age + 5;
        } else {
            BMR = 10 * weight + 6.25 * height - 5 * age - 161;
        }
        int cond = condition_judge();
        if (cond == 1) {
            return BMR * 1.2;
        } else if (cond == -1) {
            return BMR * 1.6;
        } else {
            return BMR * 1.4;
        }
    }
};